packages/manager/src/queries/accountPayment.ts

```diff
@@ -18,11 +23,14 @@ export const usePaymentMethodsQuery = (params: any = {}, filter: any = {}) => {
 };
 
 export const useAllPaymentMethodsQuery = () => {
+  const { data: grants } = useGrants();
+
   return useQuery<PaymentMethod[], APIError[]>(
     queryKey + '-all',
     getAllPaymentMethodsRequest,
     {
       ...queryPresets.oneTimeFetch,
+      enabled: grants?.global.account_access !== null,
```

Comment:
![Screen Shot 2022-07-01 at 12 40 31 PM](https://user-images.githubusercontent.com/6440455/176935878-8a81bd97-b53b-46a3-a37a-b074a97cde94.jpg)

The `!== null` seems spot on. I think we just need this optional chain to not crash non-restricted accounts

```suggestion
      enabled: grants?.global?.account_access !== null,
```